core/tests/protractor_utils/ProfilePage.js

```diff
@@ -79,16 +90,19 @@ var ProfilePage = function() {
 
   this.expectToHaveExplorationCards = async function() {
     var explorationCardsCount = await allExplorationCardElements.count();
-    if (explorationCardsCount === 0) {
-      throw new Error('There is no exploration card on this profile');
-    }
-    expect(explorationCardsCount).toBeGreaterThan(0);
+    await waitFor.visibilityOf(
+      allExplorationCardElements.first(),
+      'Exploration cards is not present or taking time to display');
+    expect(await explorationCardsCount).toBeGreaterThan(0);
```

Comment:
This is great! Now, the waitFor will fail whenever there are no cards present. That means you don't actually need `explorationCardsCount` anymore, and you don't need to check that it is greater than zero!